icu4c/source/i18n/unicode/simplenumberformatter.h

```diff
@@ -0,0 +1,331 @@
+        UErrorCode &status);
+
+    /**
+     * Creates a new SimpleNumberFormatter, overriding the grouping strategy and symbols.
+     *
+     * IMPORTANT: For efficiency, this function borrows the symbols. The symbols MUST remain valid
+     * for the lifetime of the SimpleNumberFormatter.
+     *
+     * @draft ICU 73
+     */
+    static SimpleNumberFormatter forLocaleAndSymbolsAndGroupingStrategy(
+        const icu::Locale &locale,
+        const DecimalFormatSymbols &symbols,
+        UNumberGroupingStrategy groupingStrategy,
+        UErrorCode &status);
+
+    /**
+     * Formats a value using this SimpleNumberFormatter.
+     *
+     * The SimpleNumber argument is "consumed". A new SimpleNumber object should be created for
+     * every formatting operation.
+     *
+     * @draft ICU 73
+     */
+    FormattedNumber format(SimpleNumber value, UErrorCode &status) const;
+
+    /**
+     * Formats an integer using this SimpleNumberFormatter.
+     *
+     * For more control over the formatting, use SimpleNumber.
+     *
+     * @draft ICU 73
+     */
+    FormattedNumber formatInt64(int64_t value, UErrorCode &status) const {
+        return format(SimpleNumber::forInt64(value, status), status);
+    }
+
+#ifndef U_HIDE_INTERNAL_API
+    /**
+     * Run the formatter with the internal types.
+     * @internal
+     */
+    void formatImpl(impl::UFormattedNumberData* data, USimpleNumberSign sign, UErrorCode& status) const;
+#endif // U_HIDE_INTERNAL_API
+
+    /**
+     * Destruct this SimpleNumberFormatter, cleaning up any memory it might own.
+     *
+     * @draft ICU 73
+     */
+    ~SimpleNumberFormatter() {
+        cleanup();
+    }
+
+    /**
+     * Creates a shell, initialized but non-functional SimpleNumberFormatter.
+     *
+     * @draft ICU 73
+     */
+    SimpleNumberFormatter() = default;
+
+    // FIXME: Make sure these move constructors are well tested
+
+    /**
+     * SimpleNumberFormatter: Move constructor.
+     *
+     * @draft ICU 73
+     */
+    SimpleNumberFormatter(SimpleNumberFormatter&& other) U_NOEXCEPT {
+        fGroupingStrategy = other.fGroupingStrategy;
+        fOwnedSymbols = other.fOwnedSymbols;
+        fMicros = other.fMicros;
+        fPatternModifier = other.fPatternModifier;
+        other.fOwnedSymbols = nullptr;
+        other.fMicros = nullptr;
+        other.fPatternModifier = nullptr;
+    }
+
+    /**
+     * SimpleNumberFormatter: Move assignment.
+     *
+     * @draft ICU 73
+     */
+    SimpleNumberFormatter& operator=(SimpleNumberFormatter&& other) U_NOEXCEPT {
+        cleanup();
+        fGroupingStrategy = other.fGroupingStrategy;
+        fOwnedSymbols = other.fOwnedSymbols;
+        fMicros = other.fMicros;
+        fPatternModifier = other.fPatternModifier;
+        other.fOwnedSymbols = nullptr;
+        other.fMicros = nullptr;
+        other.fPatternModifier = nullptr;
+        return *this;
+    }
+
+  private:
+    void initialize(
```

Comment:
optional: why not a private constructor with this signature?

Reply:
It has to do with whether to save something into the fOwnedSymbols field. It was easier to write it as a private initialize function, which is a pattern fairly common in icu4c